go/vt/topotools/routing_rules.go

```diff
@@ -27,6 +27,19 @@ import (
 	vschemapb "vitess.io/vitess/go/vt/proto/vschema"
 )
 
+//region routing rules
+
+func GetRoutingRulesMap(rules *vschemapb.RoutingRules) map[string][]string {
+	if rules == nil {
+		return nil
+	}
+	rulesMap := make(map[string][]string, len(rules.Rules))
+	for _, rr := range rules.Rules {
+		rulesMap[rr.FromTable] = rr.ToTables
```

Comment:
I think this is where we'd want to check that the rule is for writes/primary tables (meaning that the fromTable value has no `@[replica,rdonly]` suffix), e.g.:
```
    {
      "fromTable": "customer",
      "toTables": [
        "customer.customer"
      ]
    },
```